src/engine/renderer/glsl_source/reliefMapping_fp.glsl

```diff
@@ -41,27 +38,24 @@ vec3 NormalInTangentSpace(vec2 texNormal)
 	vec3 normal;
 
 #if defined(r_normalMapping)
-	if (u_HeightMapInNormalMap == 0)
-	{
-		// the Capcom trick abusing alpha channel of DXT1/5 formats to encode normal map
-		// https://github.com/DaemonEngine/Daemon/issues/183#issuecomment-473691252
-		//
-		// the algorithm also works with normal maps in rgb format without alpha channel
-		// but we still must be sure there is no height map in alpha channel hence the test
-		//
-		// crunch -dxn seems to produce such files, since alpha channel is abused such format
-		// is unsuitable to embed height map, then height map must be distributed as loose file
-		normal = texture2D(u_NormalMap, texNormal).rga;
-		normal.x *= normal.z;
-		normal.xy = 2.0 * normal.xy - 1.0;
-		normal.z = sqrt(1.0 - dot(normal.xy, normal.xy));
-	}
-	else
-	{
-		// alpha channel contains the height map so do not try to restore the normal map from it
-		normal = texture2D(u_NormalMap, texNormal).rgb;
-		normal = 2.0 * normal - 1.0;
-	}
+#if defined(USE_HEIGHTMAP_IN_NORMALMAP)
+	// alpha channel contains the height map so do not try to reconstruct normal map from it
+	normal = texture2D(u_NormalMap, texNormal).rgb;
+	normal = 2.0 * normal - 1.0;
```

Comment:
Don't you still need to reconstruct z in this case?

Reply:
no need to reconstruct z which is stored in `normal.b`.

Reply:
In Fuma's comment the "DXT1/BC1" case has only r and g defined, no b or a.

Reply:
Because the compute can reconstruct z without reading it. But we can just read it.
The alternative would be to reconstruct z after overwritten the heightmap with 1 to not break reconstruction.

The reconstuction trick only works for normalmap without heightmap. In one case the normalmap is stored in GBA (DXT5nm), in other case the normalmap is stored in RGB. The trick leverages the fact reading missing alpha from RGB texture gives 1.

We use the trick to avoid to do some `ifdef USE_DXT5NM` when we know there is no heigtmap, and there can't be DXT5nm textures with heightmap because alpha channel is already used (abused) so there is no need to use any trick when we know there is an heightmap, there is only one format to grok whatever the compression (or it is compressed or not).

Reply:
> DXT1/BC1 compressed texture
> Assumes R = (x+1.0)/2.0, G =(y+1.0)/2.0, B = const, A = 1.0

I don't understand. Why is this case no longer relevant? Or are you deliberately dropping support?

Reply:
There is two way to get Z, either read it from file, either reconstruct it from other components.

This is the compatibility matrix for those two methods:

|format               |storage|heightmap in alpha|compatible with trick compute|compatible with standard read|
|---------------------|-------|------------------|-----------------------------|-----------------------------|
|PNG normalMap        |RGB1   |:x:               |:heavy_check_mark:           |:heavy_check_mark:           |
|PNG normalHeightMap  |RGBA   |:heavy_check_mark:|:x:                          |:heavy_check_mark:           |
|DXT1 normalMap       |RGB1   |:x:               |:heavy_check_mark:           |:heavy_check_mark:           |
|DXT1 normalHeightMap |RGBA   |:heavy_check_mark:|:x:                          |:heavy_check_mark:           |
|DXTn normalMap       |1BGA   |:x:               |:heavy_check_mark:           |:x:                          |

NormalMap without HeightMap in alpha can can be read with trick compute or standard read whatever the format.
NormalMap with HeightMap in alpha can't be read with trick compute.

If we know the NormalMap has an HeightMap, we have to avoid the trick compute.
We can use standard read when there is an HeightMap in NormalMap since a DXTn NormalMap can't embed an HeightMap.

Reply:
You're right that the stored texture file does have a blue channel, which is a don't-care value in the case of RG normal maps. But if I understand correctly, the blue channel may be set to a constant value in order to get better compression -- the blue channel has nearly zero information, so you get a higher resolution on the red and green ones.

Reply:
That's probably the reason why people implemented DXn.

I don't see the issue in reusing blue channel when it exists instead of doing an `sqrt`. The correct way to load a normal map without trick is to read RGB.

Also, to make sure the trick works we have to properly re-normalize normalmap while compressing them. We patched crunch for that, see the _rtopmip renormalize_ commit: https://github.com/DaemonEngine/crunch/commit/902f075ac933b7d21c6de3e33cdfa484e1818b54

Well, in any way this is not really on topic about implementing loose heightmaps.

Reply:
OK I see, for now you are assuming that no one is combining the RG-only compression trick with height map in normal map.

Reply:
> OK I see, for now you are assuming that no one is combining the RG-only compression trick with height map in normal map.

Yes, since the DXTn compression uses the alpha channel for the normalmap, there is no way to store an heightmap in normalmap alpha channel.